src/main/kotlin/com/cjbooms/fabrikt/generators/controller/ControllerGeneratorUtils.kt

```diff
@@ -48,4 +49,41 @@ object ControllerGeneratorUtils {
 
     private fun httpVerbMethodName(verb: String, isSingleResource: Boolean) =
         if (isSingleResource) "${verb}ById" else verb
+
+
+    /**
+     * Enum definition for different cases of authentication
+     *
+     * N0_SECURITY when no @Secured Annotation is to be added neither an authentication input parameter
+     * AUTHENTICATION_REQUIRED when @Secured Annotation IS_AUTHENTICATED (for micronaut) is to be added and authentication as input parameter
+     * AUTHENTICATION_PROHIBITED when @Secured Annotation IS_ANONYMOUS (for micronaut) is to be added and no authentication input parameter
+     * AUTHENTICATION_OPTIONAL when @Secured Annotation IS_AUTHENTICATED and IS_ANONYMOUS (for micronaut) is to be added and authentication as optional input parameter
+     */
+    enum class SecuritySupport(val allowsAuthenticated: Boolean, val allowsAnonymous: Boolean) {
+        NO_SECURITY(false, false),
+        AUTHENTICATION_REQUIRED(true, false),
+        AUTHENTICATION_PROHIBITED(false, true),
+        AUTHENTICATION_OPTIONAL(true, true),
+    }
+
+
+    /**
+     * Differentiating the cases of authentication
+     *
+     * depending on the security settings on an operation the SecuritySupport Enum is defined
+     */
```

Comment:
**issue:** Not to be too pedantic, but please use proper casing and punctiation in comments, especially KDoc documentation and when that's the already existing style in a project :)

```kotlin
    /**
     * Computes the [SecuritySupport] of a given operation.
     * @param defaultValue The "API-global" security support to use in case the operation itself does not define any.
     */
```

Reply:
better this way?